### PR TITLE
 Don't add AddHttpContextAccessor to Startup.cs

### LINKED ISSUE #452: UA0005 failed to apply code changes
### Describe the bug
The codefixer for UA0005 failed to apply code changes.

### To Reproduce
1. Create a web library project
2. Add a Web API controller that uses HttpContext.Current
3. Run upgrade-assistant on Web Library project

### Exceptions (if any)
[15:43:58 WRN] Diagnostic UA0005 was not fixed as expected. This may be caused by the project being in a bad state (did NuGet packages restore correctly?) or by errors in analyzers or code fix providers related to this diagnostic.
[15:43:58 INF] Upgrade step Apply fix for UA0005: Do not use HttpContext.Current applied successfully

### Further technical details
- Include the output of `upgrade-assistant --version`
version '0.2.222101'

### PR DESCRIPTION
**Pull request by mjrousos**
This makes a few updates to the HttpContext.Current analyzer and code fix provider:

1.  Don't add AddHttpContextAccessor to Startup.cs. It's still nice to have that there to make it easier to use IHttpContextAccessor from DI, but it's not required by HttpContextHelper anymore and it seems best to make as few changes as possible to users' startup files.
2. Update HttpContext.Current analyzer to also identify ControllerBase.HttpContext.Current as needing updated since that's a pattern we can arrive at mid-upgrade.
3. Fix HttpContextHelper to include `using System`.

Fixes #452

### CODE REVIEW

File: src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/HttpContextCurrentAnalyzer.cs [hunk 1]
```diff
@@ -142,9 +143,20 @@ private static bool TryMatchSymbol(SyntaxNodeAnalysisContext context, SyntaxNode
                     return false;
                 }
             }
+            else if (accessedSymbol is IPropertySymbol propSymbol)
```
Comment (KSchlobohm) [inline src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/HttpContextCurrentAnalyzer.cs:146]: This looks like a bugfix to be called out in the changelog.

**Review by KSchlobohm — APPROVED** [review]
Add changelog pls, then looks good.

File: src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes/Templates/HttpContextHelper.cs [hunk 2]
```diff
@@ -1,4 +1,6 @@
-﻿namespace Microsoft.AspNetCore.Http
+﻿using System;
```
Comment (twsouthwick) [inline src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes/Templates/HttpContextHelper.cs:1]: Is this needed?

**Review by twsouthwick — APPROVED** [review]

File: src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes/Templates/HttpContextHelper.cs (hunk 2 above)
Comment (mjrousos) [reply src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes/Templates/HttpContextHelper.cs:1]: Yep. `System.ObsoleteAttribute`.

File: src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/HttpContextCurrentAnalyzer.cs (hunk 1 above)
Comment (mjrousos) [reply src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/HttpContextCurrentAnalyzer.cs:146]: Good call. I'll update the changelog.

**Pull request merged**